File: Components/DevExpressVCL/Demos/ExpressEditors Library/Delphi/EditorsLookupDemo/EditorsLookupDemoNewUser.pas
unit EditorsLookupDemoNewUser;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms,
  Dialogs, cxHyperLinkEdit, cxDBEdit, cxMaskEdit, cxControls, cxContainer,
  cxEdit, cxTextEdit, StdCtrls, cxButtons, cxDropDownEdit, cxLookupEdit,
  cxDBLookupComboBox, cxLookAndFeelPainters, cxDBLookupEdit;

type
  TEditorsLookupDemoNewUserForm = class(TForm)
    edFirstName: TcxDBTextEdit;
    edMidleName: TcxDBTextEdit;
    edLastName: TcxDBTextEdit;
    edCountry: TcxDBTextEdit;
    edCity: TcxDBTextEdit;
    mePostalCode: TcxDBMaskEdit;
    edAddress: TcxDBTextEdit;
    mePhone: TcxDBMaskEdit;
    meFax: TcxDBMaskEdit;
    heEMail: TcxDBHyperLinkEdit;
    heHomePAge: TcxDBHyperLinkEdit;
    Label1: TLabel;
    Label2: TLabel;
    Label3: TLabel;
    Label4: TLabel;
    Label5: TLabel;
    Label6: TLabel;
    Label7: TLabel;
    Label8: TLabel;
    Label9: TLabel;
    Label10: TLabel;
    Label11: TLabel;
    lcbDepartment: TcxDBLookupComboBox;
    Label12: TLabel;
    btnOK: TcxButton;
    btnCancel: TcxButton;
    procedure FormClose(Sender: TObject; var Action: TCloseAction);
  private
    { Private declarations }
  public
    { Public declarations }
    function ShowEx(AName: String): Integer;
  end;

var
  EditorsLookupDemoNewUserForm: TEditorsLookupDemoNewUserForm;

implementation

uses EditorsLookupDemoData, DB;   


{$R *.dfm}

function TEditorsLookupDemoNewUserForm.ShowEx(AName: String): Integer;
var
  LName: string;
  APos: Integer;

  procedure DeleteSpaces;
  begin
    AName := Trim(AName);
    repeat
      APos := Pos('  ', AName);
      while (APos <> 0) and (AName[APos + 1] = ' ') do
        Delete(AName, APos + 1, 1);
    until APos = 0
  end;
begin
  DeleteSpaces;
  APos := Pos(' ', AName);
  if APos <> 0 then
  begin
    LName := Copy(AName, APos + 1, Length(AName) - APos);
    AName := Copy(AName, 1, APos - 1);
  end;
  with EditorsLookupDemoDataDM do
  begin
    tblUsers.Append;
    tblUsersFNAME.AsString := AName;
    tblUsersLNAME.AsString := LName;
    Result := ShowModal;
  end;

end;

procedure TEditorsLookupDemoNewUserForm.FormClose(Sender: TObject;
  var Action: TCloseAction);
begin
  with EditorsLookupDemoDataDM do
  if ModalResult = mrOK then tblUsers.Post
  else tblUsers.Cancel;
end;

end.

File: Components/DevExpressVCL/Demos/ExpressEditors Library/CBuilder/EditorsLookupDemo/EditorsLookupDemoNewUser.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "EditorsLookupDemoNewUser.h"
#include "EditorsLookupDemoData.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxButtons"
#pragma link "cxContainer"
#pragma link "cxControls"
#pragma link "cxDBEdit"
#pragma link "cxDBLookupComboBox"
#pragma link "cxDropDownEdit"
#pragma link "cxEdit"
#pragma link "cxHyperLinkEdit"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxLookupEdit"
#pragma link "cxMaskEdit"
#pragma link "cxTextEdit"
#pragma link "cxDBLookupEdit"
#pragma resource "*.dfm"
TEditorsLookupDemoNewUserForm *EditorsLookupDemoNewUserForm;
//---------------------------------------------------------------------------
__fastcall TEditorsLookupDemoNewUserForm::TEditorsLookupDemoNewUserForm(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TEditorsLookupDemoNewUserForm::FormClose(TObject *Sender, TCloseAction &Action)
{
  if (ModalResult == mrOk) EditorsLookupDemoDataDM->tblUsers->Post();
  else EditorsLookupDemoDataDM->tblUsers->Cancel();
}
//---------------------------------------------------------------------------

int TEditorsLookupDemoNewUserForm::ShowEx(String AName)
{
  DeleteSpaces(AName);
  int APos = AName.Pos(" ");
  String LName;
  if (APos != 0){
    LName = AName.SubString(APos + 1,AName.Length() - APos);
    AName = AName.SubString(1, APos - 1);
  }
  EditorsLookupDemoDataDM->tblUsers->Append();
  EditorsLookupDemoDataDM->tblUsersFNAME->AsString = AName;
  EditorsLookupDemoDataDM->tblUsersLNAME->AsString = LName;
  return ShowModal();
}
//---------------------------------------------------------------------------

void TEditorsLookupDemoNewUserForm::DeleteSpaces(String &AStr)
{
  AStr = Trim(AStr);
  int APos;
  while ((APos = AStr.Pos("  ")) != 0)
    AStr.Delete(APos + 1, 1);
}
//---------------------------------------------------------------------------


